### PR TITLE
UIF-503 Check workflow status of order in the encumbrance pane actions

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://folio-org.atlassian.net/browse/UIF-503

Init PR: #785 

In the preceding PR (#785), the workflow status of the order was assessed based on the source's transaction data. Nonetheless, this data lacks the most recent order updates, which necessitates additional order information to be loaded for encumbrances.

## Approach

Use the `react-query` hook to fetch the order by ID from the encumbrance source order.

## Screenshot

https://github.com/folio-org/ui-finance/assets/88109087/ab0878a0-033b-4e13-b1f2-c92faecaa1aa

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/hooks/useOrder/useOrder.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  useOkapiKy,
+} from '@folio/stripes/core';
+
+import { ORDERS_API } from '@folio/stripes-acq-components';
+
+export const useOrder = (orderId) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'purchase-order' });
+
+  const { isLoading, data } = useQuery({
```
Comment (OleksandrHladchenko1) [inline src/common/hooks/useOrder/useOrder.js:14]:
```suggestion
  const {
    isLoading,
    data,
  } = useQuery({
```

File: src/common/hooks/useOrder/useOrder.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { QueryClient, QueryClientProvider } from 'react-query';
```
Comment (OleksandrHladchenko1) [inline src/common/hooks/useOrder/useOrder.test.js:1]:
```suggestion
import {
  QueryClient,
  QueryClientProvider,
} from 'react-query';
```

File: src/common/hooks/useOrder/useOrder.test.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import { renderHook, waitFor } from '@folio/jest-config-stripes/testing-library/react';
```
Comment (OleksandrHladchenko1) [inline src/common/hooks/useOrder/useOrder.test.js:3]: the same here

**Review by OleksandrHladchenko1 — APPROVED** [review]

File: src/common/hooks/useOrder/useOrder.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export const useOrder = (orderId) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'purchase-order' });
+
+  const {
+    data,
+    isLoading,
+  } = useQuery({
+    queryKey: [namespace, orderId],
+    queryFn: async () => ky.get(`${ORDERS_API}/${orderId}`).json(),
```
Comment (NikitaSedyx) [inline src/common/hooks/useOrder/useOrder.js:19]: you need permissions to load

Comment (usavkov-epam) [reply src/common/hooks/useOrder/useOrder.js:19]: Good point. Moreover, this requires a new okapi interface (`orders`).

**Comment by usavkov-epam** [comment]
Hello @folio-org/fe-tl-reviewers ! Could you please take a look at the PR? Thanks!

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**